js/app.js

```diff
@@ -223,3 +223,55 @@ function formatString(str, args) {
     }
     return str;
 }
+
+// VARIANT 1
+// e.g. formatString("Placeholder {0} and {1}", "Value 1", "Value 2");
+function formatString(format) {
+    for(let i = 1; i < arguments.length; i++) {
+        format = format.replace(`{${i-1}}`, arguments[i]);
+    }
+    return format;
+}
+
+// VARIANT 2
+// e.g. formatString("Placeholder {0} and {1}", "Value 1", "Value 2");
+function emplace(format, value, index) {
+    return format.replace(`{${index}}`, value);
+}
+function formatString(format) {
+    for(let i = 1; i < arguments.length; i++) {
+        format = emplace(format, arguments[i], i);
+    }
+    return format;
+}
+
+// VARIANT 3
+// e.g. formatString("Placeholder {0} and {1}", ["Value 1", "Value 2"]);
+function formatString(format, args) {
+    return args.reduce(function(result, value, index) {
+        return result.replace(`{${index}}`, value);
+    }, format);
+}
+
+// VARIANT 4
+// e.g. formatString("Placeholder {0} and {1}", "Value 1", "Value 2");
+function formatString() {
+    return Array.from(arguments).reduce(function(result, value, index) {
+        return result.replace(`{${index}}`, value);
+    });
+}
+
+
```

Comment:
I am not sure how you would pass and use the string to be formatted in these two variants (5/6)

Reply:
The usage example is shown in the comment above each one. All examples use the same format string syntax that you originally proposed.